src/org/labkey/test/tests/component/EditableGridTest.java

```diff
@@ -434,6 +482,717 @@ public void testExpandedPasteIntoSkinnySelection() throws Exception
+
+        log("Reset the grid.");
+        editableGrid.selectAll(true);
+        editableGrid.clickRemove();
+
+        log("Paste in a multi-line string without putting the cell into edit mode.");
+        editableGrid.addRows(1);
+        editableGrid.pasteFromCell(0, PASTE_ML, sbPasteString.toString());
+
+        checker().verifyTrue("Each line should have created a new row.",
+                waitFor(()->expectedValues.size() == editableGrid.getRowCount(), 1_000));
+
+        checker().verifyEquals(String.format("Values in column '%s' not as expected.", PASTE_ML),
+                expectedValues, editableGrid.getColumnData(PASTE_ML));
+
+        checker().screenShotIfNewError("Paste_Into_Multiple_Cells_Error");
+
+    }
+
+    /**
+     * <p>
+     *     Test the sizing of the multi-line edit cell in an editable grid.
+     * </p>
+     * <p>
+     *     This test will:
+     *     <ul>
+     *         <li>Many long lines will size the textArea to the maximum allowed size.</li>
+     *         <li>Hitting enter removes focus and the updated text is saved.</li>
+     *         <li>One long line (no new line) expands only the width and does not change the heigth.</li>
+     *         <li>Clicking outside of the textArea save the changes.</li>
+     *         <li>Many shot lines expands only the height and not the width.</li>
+     *         <li>Hitting the 'ESC' key closes the textArea and does not save the changes.</li>
+     *     </ul>
+     * </p>
+     */
+    @Test
+    public void testMultiLineCellSizing()
+    {
+
+        EditableGrid editableGrid = goToEditableGrid(PASTING_SAMPLE_TYPE);
+        editableGrid.addRows(1);
+
+        // Scroll the last column into view it will make any failure screenshots more useful.
+        String lastColumnName = editableGrid.getColumnNames().get(editableGrid.getColumnNames().size() - 1);
+        scrollIntoView(editableGrid.getCell(0, lastColumnName));
+
+        int emptyWidth = editableGrid.getCell(0, PASTE_ML).getSize().getWidth();
+        int emptyHeight = editableGrid.getCell(0, PASTE_ML).getSize().getHeight();
+
+        int maxWidth = 600;
+        int maxHeight = 205;
+
+        log("Test entering several long lines.");
+
+        StringBuilder longLine01 = new StringBuilder();
+        StringBuilder longLine02 = new StringBuilder();
+        StringBuilder longLine03 = new StringBuilder();
+
+        for(int i = 0; i < 250; i++)
+        {
+            longLine01.append("a");
+            longLine02.append("b");
+            longLine03.append("c");
+        }
+
+        WebElement gridCell = editableGrid.getCell(0, PASTE_ML);
+        log("Cell initial Width & Height: " + gridCell.getSize());
+        log("Empty width: " + emptyWidth + " empty height: " + emptyHeight);
+
+        WebElement editCell = editableGrid.doubleClickMultiLineCell(0, PASTE_ML);
+
+        Actions actions;
+
+        for(int rows = 0; rows < 50; rows++)
```

Comment:
Why 50 times?

Reply:
No particular reason. 
Reduced it to 10, for a total of 30 lines to be added.